cranelift/isle/docs/language-reference.md

```diff
@@ -1554,6 +1554,7 @@ The grammar accepted by the parser is as follows:
 
 <spec-expr> ::= <int>
               | <spec-bv>
+              | "result"
```

Comment:
This is considered an identifier and is not a separate keyword.

Reply:
If it is an integral part of the verification and if produced value only bound to ident with the exact spelling "result", I think it deserves a place in the spec. Your call :))

Reply:
It's currently special, but the intent has never really been to build up a set of keywords -- in the future we may find a better way. I don't see it quite the same as a privileged name like `this` in C++ or `self` in Rust, for example.

I think you may be placing too much weight on the BNF here, too: the purpose of BNF is usually not to give a parse tree with the full semantics of a language embedded at the parse level, but rather to describe the accepted language syntax accurately and give names to concepts that can then be given meaning by higher levels of the spec. That's more or less what is going on here.

Reply:
I see. I certainly did and shouldn't. Thanks for pointing that out!